api/v1alpha1/api.go

```diff
@@ -442,7 +448,27 @@ type BackendSecurityPolicyAPIKey struct {
 	SecretRef *gwapiv1.SecretObjectReference `json:"secretRef"`
 }
 
-// BackendSecurityPolicyAWSCredentials contains the supported authentication mechanisms to access aws
+// BackendSecurityPolicyAzureCredentials contains the supported authentication mechanisms to access Azure.
+type BackendSecurityPolicyAzureCredentials struct {
```

Comment:
needs additional CEL validation tests tested in tests/crdcel/main_test.go for this construct